engines/rancher_on_aks/app/jobs/rancher_on_aks/vcpu_quota_check_job.rb

```diff
@@ -0,0 +1,22 @@
+module RancherOnAks
+  class VcpuQuotaCheckJob < ApplicationJob
+    queue_as :default
+
+    def perform(check_id:)
+      check = PreFlight::Check.find(check_id)
+      cluster_size = RancherOnAks::ClusterSize.new
```

Comment:
This could be moved inside of the initializer of `Azure::VcpuQuota`, so

```ruby
module Azure
  class VcpuQuota < Azure::Quota
    def initialize
       cluster_size = RancherOnAks::ClusterSize.new
       instance_type = cluster_size.instance_type,
       quantity = cluster_size.instance_count
     end
     ...
```
then use it in line 9 (line 8 can be removed or stay)
```ruby
check.passed =  Azure::VcpuQuota.new.sufficient_capacity?
```
My suggestion is that if `VcpuQuota` needs the instance type and instance count from Cluster class,
maybe it shoud go there and `VcpuQuotaCheckJob` class does not need to know about it

Reply:
The VcpuQouta class belongs to the Azure module; the ClusterSize is part of the RancherOnAks workflow. In the case of a different workflow, I don't want to pollute the a class in the Azure engine with a dependency on the RancherOnAks workflow engine. As a result the placement becomes obvious, the VcpuQuotaCheckJob, which leverages the RancherOnAks::ClusterSize, is part of the RancherOnAks engine. Any other workflow could independently implement a VcpuQuotaCheckJob using some other instance type source, and still check with the Azure::VcpuQuota class.